util/unzip.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package util
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/keybase/go-logging"
+)
+
+// UnzipOver safely unzips a filename to a destination path. If destination
+// path exists, it will be removed first. The filename must have a ".zip"
+// extension.
+func UnzipOver(filename string, destinationPath string, log logging.Logger) error {
+	if !strings.HasSuffix(filename, ".zip") {
+		return fmt.Errorf("File has unsupported extension: %q", filename)
+	}
+
+	if destinationPath == "" {
+		return fmt.Errorf("Invalid destination %q", destinationPath)
+	}
+
+	if _, ferr := os.Stat(destinationPath); ferr == nil {
```

Comment:
so this'll remove it then try and unzip over it, is this dangerous? don't we want something like unzip into a temp, then do a name swap?